bot/utils/transformData.ts

```diff
@@ -0,0 +1,98 @@
+import { poolpairs, stats } from '@defichain/whale-api-client';
+import { BigNumber } from 'bignumber.js';
+
+import { StateRelayer } from '../../generated';
+import { MasterNodeData, PairData, VaultData } from './types';
+
+const DECIMALS = 18;
+const DENOMINATION = 'USDT';
+
+type AsyncReturnType<T extends (...args: any) => any> = T extends (...args: any) => Promise<infer R> ? R : any;
+type StatsData = AsyncReturnType<typeof stats.Stats.prototype.get>;
+type PoolPairData = AsyncReturnType<typeof poolpairs.PoolPairs.prototype.list>;
+type DexPriceResult = AsyncReturnType<typeof poolpairs.PoolPairs.prototype.listDexPrices>;
+
+const transformToBigInt = (str: string, decimals: number): bigint =>
+  BigInt(new BigNumber(str).multipliedBy(new BigNumber('10').pow(decimals)).toFixed(0, 1));
+
+export function transformDataVault(statsData: StatsData): VaultData {
+  const dataVault = {} as VaultData;
+
+  const totalLoanValue = statsData.loan.value.loan;
+  const totalCollateralValue = statsData.loan.value.collateral;
+  dataVault.noOfVaultsNoDecimals = transformToBigInt(statsData.loan.count.openVaults.toString(), 0);
+  dataVault.totalLoanValue = transformToBigInt(totalLoanValue.toString(), DECIMALS);
+  dataVault.totalCollateralValue = transformToBigInt(totalCollateralValue.toString(), DECIMALS);
+  dataVault.totalCollateralizationRatio = transformToBigInt(
+    ((totalCollateralValue / totalLoanValue) * 100).toFixed(3).toString(),
+    DECIMALS,
+  );
+  dataVault.activeAuctionsNoDecimals = transformToBigInt(statsData.loan.count.openAuctions.toString(), 0);
+
+  return dataVault;
+}
+
+export function transformDataMasternode(statsData: StatsData): MasterNodeData {
+  const dataMasterNode = {} as MasterNodeData;
+  dataMasterNode.totalValueLockedInMasterNodes = transformToBigInt(statsData.tvl.masternodes.toString(), DECIMALS);
+  dataMasterNode.zeroYearLockedNoDecimals = transformToBigInt(statsData.masternodes.locked[0].count.toString(), 0);
+  dataMasterNode.fiveYearLockedNoDecimals = transformToBigInt(statsData.masternodes.locked[2].count.toString(), 0);
+  dataMasterNode.tenYearLockedNoDecimals = transformToBigInt(statsData.masternodes.locked[1].count.toString(), 0);
```

Comment:
Hmm, so the API given does not assure such order?

Reply:
> Hmm, so the API given does not assure such order?
hmm, actually that's my point, haha. If jellyfish changes it for some reason, then the bot will be incorrect. So, what if instead of assuming the lockedInYears via index, why not via `weeks`?

Reply:
fixed!